crates/transparency/src/map/node.rs

```diff
@@ -59,27 +74,59 @@ impl<D: SupportedDigest> Node<D> {
     /// Returns:
     ///   * the new node that must replace the current node.
     ///   * whether or not this is a new entry in the map.
-    pub fn insert(&self, path: &mut Path<D>, leaf: Hash<D>) -> (Self, bool) {
+    pub fn insert(&self, path: &mut Path<'_, D>, value: Hash<D>) -> (Self, bool) {
```

Comment:
Rust might let you remove this lifetime on `Path` since it's the elided lifetime `'_`.

Reply:
Can remove it, but `cargo check` still warns

Reply:
weird, wouldn't guess that it'd be legal but a warning. no worries